extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -864,10 +864,14 @@ public Uni<SecurityIdentity> apply(final AuthorizationCodeTokens tokens, final T
                             if (isIdTokenRequired(configContext)) {
                                 LOG.errorf("ID token is not available in the authorization code grant response");
                                 return Uni.createFrom().failure(new AuthenticationCompletionException());
-                            } else {
+                            } else if (tokens.getAccessToken() != null) {
                                 tokens.setIdToken(generateInternalIdToken(configContext, null, null,
                                         tokens.getAccessTokenExpiresIn()));
                                 internalIdToken = true;
+                            } else {
+                                LOG.errorf(
+                                        "Neither ID token nor access tokens are available in the authorization code grant response");
```

Comment:
Is there anything the dev can do to fix that? (Check some URL, setting, etc?)

Reply:
Hi @gastaldi It depends on the situation, for example, in this case we say GitHub returning GitHub specific error in JSON format with `200` response type... 
Perhaps I should add `Please check the logs for more details` ?

Reply:
That would be nice. Also if the data isn't visible in the logs, maybe point to a URL where you can find more info on how to fix that

Reply:
That is in the actual GitHub response, it includes link to the trouble shooting guide, the problem they return it with 200 which needs a debug level to be seen... Let me tune it a bit

Reply:
I gave it a try :-)